### PR TITLE
aw-16: add lead details page

### PR DESCRIPTION
**Pull request by canterbery**
## Describe your changes

- Added detailed lead view , matching properties with infinite scroll.  Added implementation of 3 filters to main page property filters

image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-180)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AKutusenko — APPROVED** [review]

**Review by rudenkoserhii — APPROVED** [review]

**Comment by lyudad** [comment]
Screenshot 2023-12-18 at 19 57 12

**Comment by canterbery** [comment]
> Screenshot 2023-12-18 at 19 57 12

image

**Comment by lyudad** [comment]
fonts  still have different sizes

**Comment by canterbery** [comment]
> fonts still have different sizes

No they don't

image

**Comment by lyudad** [comment]
Screenshot 2023-12-20 at 22 37 54

**Comment by canterbery** [comment]
image

it's the same font

**Comment by lyudad** [comment]
Screenshot 2023-12-22 at 10 57 23
Screenshot 2023-12-22 at 10 57 43

**Comment by canterbery** [comment]
There were only functional filter changes. Not in UI. UI was added before so here I implemented logic for several filter options.

File: src/sections/lead/lead-details-view.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  const scrollToTop = (): void => {
+    window.scrollTo({ top: 0, behavior: 'smooth' });
+  };
+
+  const { data } = leadDetails;
+
+  const contact = data.contact_name_contact;
+  const workflow = data.conversion_status_workflow_stage;
+  const propertyType = data.property_type;
```
Comment (lyudad) [inline src/sections/lead/lead-details-view.tsx]: почему ты на строке 27 не доразвернешь обьект, зачем 29 - 31 строка

**Pull request merged**